components/kyma-environment-broker/internal/process/upgrade_kyma/internal_eval_updater.go

```diff
@@ -0,0 +1,31 @@
+package upgrade_kyma
+
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/avs"
+	"github.com/sirupsen/logrus"
+)
+
+type InternalEvalUpdater struct {
+	delegator *avs.Delegator
+	assistant *avs.InternalEvalAssistant
+	avsConfig avs.Config
+}
+
+func NewInternalEvalUpdater(delegator *avs.Delegator, assistant *avs.InternalEvalAssistant, config avs.Config) *InternalEvalUpdater {
```

Comment:
I suggest to refactor this to `EvalUpdater` and add logic to also turn the status of the external evaluation.

So one function to set status for both the internal and external eval, and another function to restore the status for both of them.

Reply:
It does makes sense, I completely forgot about the fact that external monitors will still ping the systems during the upgrade.